microprofile/WFLY-12901_MicroProfile_JWT_Quickstart.adoc

```diff
@@ -0,0 +1,123 @@
+
+The general steps covered will be: -
+
+1. Generation of a new Maven project using the `archetype:gnerate` goal and a released WildFly archetype.
+
+This step may include some tweaks to the new project to make it suitable for the following steps.
+
+2. Ensuring the project has a base JAX-RS endpoint, deploy the project and call the endpoint using `curl`.
+
+At this stage the project is a simple JAX-RS deployment without security ready for the subsequent steps to 
+enable MicroProfile JWT based security.  The initial endpoint should report the name of the current `Principal` 
+which for now will be `anonymous`. 
+
+3. Generate a suitable public / private key.
+
+As a Quickstart this will be making use of locally generated tokens so local keys will be required.
+
+4. Add a utility to the project to generate Base64 encoded JWT tokens along with the ability to customise some 
+of the claims.
+
+It will be possible when executing this utility to specify the name of the principal, the birthday of the principal,
+and the groups to assign.  The birthday claim will be used to demonstrate claim injection later within the quickstart.
+
+The user will be able to execute the token utility with the `exec:java` Maven goal.
+
+5. The minimal steps will be performed to activate the `MP-JWT` authentication mechanism on the endpoint.
+
+This will cover aspects such as ensuring CDI is enabled, adding the `@LoginConfig` annotation and the relevant 
+MicroProfile Config properties needed to configure JWT.
+
+The command line tool `curl` will be used to call the endpoint, this time using a generated token to verify 
+the token is accepted and confirm the correct principal name is returned.
+
+6. Role based Authorization will be enabled using a new endpoint.
+
+So the endpoint used so far is left unmodified a new endpoint will be added which now uses the `@RolesAllowed`
+annotation to specify the required roles.
+
+The token utility added previously can be used to generate tokens with and without the required role to test the
+behaviour.
+
+7. Claim Injection using a new endpoint.
+
+One futher endpoint will be added, this endpoint will make use of an injected birthday claim and will report to 
+the caller how long it is until their birthday.
+
+This endpoint again will be called using tokens generated by the token generation utility.
+
+
+The text for each of these steps is the text that will form the README of the quickstart.
+
+The code which results from executing these steps is the code that will be added to the Quickstarts project as a new 
+sub-module.  The code will not be modified further, however for long term maintenance it may be required to set the module's pom to use the Quickstart parent pom to inherit dependency versioning.
+
+=== Nice-to-Have Requirements
+
+None
+
+=== Non-Requirements
+
+Anything not listed under Hard Requirements.
+
+== Test Plan
```

Comment:
This section should include information on how this quickstart will be tested. Please add the test plan here.